src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicy.cs

```diff
@@ -141,5 +149,23 @@ public override string ToString()
             builder.Append("}");
             return builder.ToString();
         }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="CorsPolicy"/>.
+        /// </summary>
+        public CorsPolicy()
+        {
+            // Regenerate 'RegularOrigins' every time the 'Origins' has changed.
+            Origins.CollectionChanged += (sender, e) =>
```

Comment:
> Are policies all generated and set during configuration?

No! We can change origins out of the configuration... When you need dynamic origins, you can write a custom service divided from `CorsService`... I have a project that I change origins dynamically in `EvaluateRequest` & `EvaluatePreflightRequest` with access to `policy.Origins`...
So, by using `ObservableCollection` we can generate `RegularOrigins` each time is needed.